src/Symfony/Bridge/PhpUnit/AssertDeprecationTrait.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit;
+
+use PHPUnit\Framework\Assert;
+
+trait AssertDeprecationTrait
+{
+    /**
+     * @param callable(): TReturn $callable
+     * @return TReturn
+     * @template TReturn
+     */
+    private static function assertDeprecation(string $expectedMessage, callable $callable): mixed
+    {
+        $matched = false;
+        $observed = [];
+        $previousErrorHandler = null;
+        $previousErrorHandler = set_error_handler(static function (int $type, string $message) use (&$previousErrorHandler, &$matched, &$observed, $expectedMessage) {
+            if (($type === E_USER_DEPRECATED || $type === E_DEPRECATED)) {
+                if (str_contains($message, $expectedMessage)) {
+                    return $matched = true;
+                }
+
+                $observed[] = $message;
+            }
+
+            return $previousErrorHandler(...func_get_args());
+        }) ?? static function () {
```

Comment:
[nitpicking] `static fn () => false` ?

Reply:
We still maintain compatibility with PHP 7.2 in this package.